ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/activity/PaywallActivityLauncher.kt

```diff
@@ -31,6 +33,8 @@ interface PaywallDisplayCallback {
  */
 class PaywallActivityLauncher(resultCaller: ActivityResultCaller, resultHandler: PaywallResultHandler) {
     private val activityResultLauncher: ActivityResultLauncher<PaywallActivityArgs>
+    private val weakActivity = WeakReference(resultCaller as? Activity)
+    private val weakFragment = WeakReference(resultCaller as? Fragment)
```

Comment:
Honestly, I believe this is very hacky... But I wasn't able to figure out how to obtain a reference to the activity otherwise that would let us check if it was finishing from here... 

This should work most times since usually the `ActivityResultCaller` is normally the activity or a fragment 

I was thinking an alternative would be to remove this `PaywallActivityLauncher`, expose the `PaywallActivity` and have the developer manage the presentation/result but I don't really like that. Open to ideas...

Reply:
I don't think this is _that_ bad to be honest. We have a requirement to check whether the Activity is still alive, but if we keep a hard reference we cause a memory leak. I think `WeakReference` is the only option.

Reply:
kinda reminds me of this TBH https://github.com/RevenueCat/purchases-ios/blob/main/Sources/Purchasing/Purchases/PurchasesOrchestrator.swift#L73-L78, I don't think it's terrible as long as its under our control. 

But definitely add comments here so future you doesn't hate present you for doing this and not explaining why